nengo_loihi/builder/inputs.py

```diff
@@ -85,3 +91,116 @@ class PESModulatoryTarget:
     def __init__(self, target):
         super().__init__()
         self.error_target = target
+
+
+class DVSFileChipNode(ChipReceiveNeurons):
+    """Node for DVS input to Loihi chip from a pre-recorded file
+
+    Parameters
+    ----------
+    file_path : string
+        The path of the file to read from. Can be a ``.aedat`` or ``.events`` file.
+    kind : "aedat" or "events" or None, optional
+        The format of the file. If ``None`` (default), this will be detected from the
+        file extension.
+    t_start : float, optional
+        Offset for the time in the file to start at, in seconds.
+    rel_time : bool, optional
+        Whether to make all times relative to the first event, or not. Defaults
+        to True for ``.aedat`` files and False otherwise.
+    pool : (int, int), optional
+        Number of pixels to pool over in the vertical and horizontal
+        directions, respectively.
+    channels_last : bool, optional
+        Whether to make the channels (i.e. the polarity) the least-significant
+        index (True) or the most-significant index (False).
+    """
+
+    def __init__(
+        self,
+        file_path,
+        format=None,
+        t_start=0,
+        rel_time=None,
+        pool=(1, 1),
+        channels_last=True,
+        label=None,
+    ):
+        self.file_path = file_path
+        assert os.path.exists(self.file_path)
+        self.format = format
+        self.t_start = t_start
+        self.rel_time = rel_time
+
+        self.dvs_height = 180
+        self.dvs_width = 240
+        self.dvs_polarity = 2
+        self.channels_last = channels_last
+        self.pool = pool
+
+        self.height = int(np.ceil(self.dvs_height / self.pool[0]))
+        self.width = int(np.ceil(self.dvs_width / self.pool[1]))
+        self.polarity = self.dvs_polarity
+        self.size = self.height * self.width * self.polarity
+
+        # for outputting images if the node is run in Nengo
+        output = DVSFileImageProcess(self)
```

Comment:
So my first inclination was that this makes a lot of sense, but when I started doing it, it made less sense.

The `DVSFileImageProcess` isn't something that we want users to be instantiating themselves. It's basically just a way to do DVS input as frames in Nengo core, and do it with the right `dt` (as opposed to modifying `.update`, which doesn't know `dt`). We always want users to use `DVSFileChipNode`, since that works properly in both Nengo core and NengoLoihi (if they did it with `DVSFileImageProcess`, and then started using the NengoLoihi simulator, it would make frames on the host, rather than just sending spikes directly to the chip, which is almost certainly not what they want).

I _guess_ it could be possible that a user would want a process to turn a DVS file into frames that they can run independently as a process, rather than having it as the output to a Node. That's the one case where I can think it would be useful for a user to make a `DVSFileImageProcess`.

I think we should make `DVSFileImageProcess` either hidden or a member of `DVSFileChipNode` (or both), to indicate that it's not meant to be instantiated by users. I would do that whether we move all the DVS parameters to `DVSFileImageProcess` or not. And I think that if we do that, then it's not really a problem to have the reference back to the node, so I wouldn't bother changing things. (The change means that you end up having all the DVS arguments on `DVSFileChipNode.__init__` and then passing them straight through to `DVSFileImageProcess.__init__`.)

Reply:
If we made the other change I suggested above (checking for the Process in the splitter, rather than the node type), then we wouldn't have that issue right? Because it'd be fine if users didn't use `DVSFileChipNode`, just a regular `Node(DVSFileImageProcess)` would work in NengoCore or NengoLoihi.